### PR TITLE
Spcr 466

### PR DESCRIPTION
**Pull request by jennikate**
### AC / Description of changes

- create PageContainer to hold layout, ErrorSummary, LoadingSpinner
- add a 500 error handler to getData hook
- update pages to use new PageContainer & 500 error
- - Voyage departure
- - Voyage arrival
- - Voyage people selector
- - Voyage people manifest
- - Voyage employee check (is anyone employed)
- - Voyage employee select (select who is employed)

 

### To Test (Developer)

in the getData hook, change the `.then(() => {})` response to be 
```
.then(() => {
  return ({
    errors: true, status: 500, message: 'There has been a problem please try again later or view the help page', id: 'helpError',
  });
});
```

- view the page
- > the page should show the page level error with help text rather than page data
 _this is the easiest way to see visually what will happen as it is very hard to mock a 500 error to the browser_

- change it back
- run tests
- > there should be a 500 test for each page that has been updated and it should pass

### Notes

**Skipped (reason in bullet)**

- voyage pleasure craft : needs refactor to a single component to make this work properly
- voyage cya : complex page needs thinking

**Not done yet**

- Pleasure craft page
- Pleasure craft form 
- People page
- People form
- Account page
- Account form
- Dashboard
- Manage your voyage reports
- POST/PATCH/DELETE calls

---

**Remember to merge to feature branches not `master`.**

### CODE REVIEW

File: src/pages/voyage/ArrivalForm.jsx [hunk 1]
```diff
@@ -32,30 +31,34 @@ const ArrivalForm = () => {
...
-    setFormData({
-      ...resp.data, ...newDateTime, ...departureDate, ...departureTimeFrom, ...departureTimeTo,
-    });
+      setFormData({
+        ...resp.data, ...newDateTime, ...departureDate, ...departureTimeFrom, ...departureTimeTo,
+      });
+    } else if (resp) {
+      setErrors({ [resp.id]: resp.message });
+    }
     setIsLoading(false);
```
Comment (Adele-B) [inline src/pages/voyage/ArrivalForm.jsx:62]: Due to the positioning of the `setIsLoading(false)` here, for some reason, it loads the arrival form with the error message above when the form shouldn't be visible

File: src/pages/voyage/PeopleSelectorForm.jsx [hunk 2]
```diff
@@ -110,121 +118,119 @@ const PeopleSelectorForm = () => {
...
-                          One or more of your saved people&apos;s travel documents has expired. You must&nbsp;
-                          <Link to="/people">update their details</Link>
-                          &nbsp;before you can add them to a voyage.
-                        </strong>
-                      </div>
-                    )}
-                    <div id="people-on-board-hint" className="govuk-hint">
-                      Select all that apply.
-                    </div>
+    // <div className="govuk-width-container ">
```
Comment (Adele-B) [inline src/pages/voyage/PeopleSelectorForm.jsx:121]: Was this commented out code supposed to be deleted?

File: src/pages/voyage/SkipperDetailsForm.jsx [hunk 3]
```diff
@@ -57,8 +56,12 @@ const SkipperDetailsForm = () => {
...
   const getVoyageData = async (id) => {
-    const response = await getData(`${VOYAGE_REPORT_URL}/${id}`);
-    setFormData(response.data);
+    const resp = await getData(`${VOYAGE_REPORT_URL}/${id}`);
+    if (resp && resp.status === 200) {
+      setFormData(resp.data);
+    } else if (resp) {
+      setErrors({ [resp.id]: resp.message });
+    }
     setIsLoading(false);
```
Comment (Adele-B) [inline src/pages/voyage/SkipperDetailsForm.jsx:65]: Due to the positioning of the `setIsLoading(false)` here, the skipper page is doing the same as the arrival page

**Review by Adele-B — CHANGES_REQUESTED** [review]

File: src/pages/voyage/PeopleSelectorForm.jsx (hunk 2 above)
Comment (jennikate) [reply src/pages/voyage/PeopleSelectorForm.jsx:121]: Yes! Deleting that :)

File: src/pages/voyage/ArrivalForm.jsx (hunk 1 above)
Comment (jennikate) [reply src/pages/voyage/ArrivalForm.jsx:62]: 👍

File: src/pages/voyage/SkipperDetailsForm.jsx [hunk 4]
```diff
@@ -74,120 +77,113 @@ const SkipperDetailsForm = () => {
...
-                    name="responsibleContactNo"
-                    type="text"
-                    value={formData.responsibleContactNo || ''}
-                    onChange={(handleChange)}
-                  />
-                </div>
+    <PageContainer errors={errors} isLoading={isLoading}>
+      {!isLoading && (
+        <form>
+          <div id="responsibleGivenName" className={`govuk-form-group ${errors.responsibleGivenName ? 'govuk-form-group--error' : ''}`}>
```
[the remark this answers is not in the document]
Comment (jennikate) [reply src/pages/voyage/SkipperDetailsForm.jsx:83]: 👍

File: src/pages/voyage/SkipperDetailsForm.jsx (hunk 3 above)
Comment (jennikate) [reply src/pages/voyage/SkipperDetailsForm.jsx:65]: 👍

**Comment by jennikate** [comment]
Splitting into two as need confirmation on whether handling 500's or persisting data is more important as a first step

**Pull request closed without merging**